### PR TITLE
Extend_Euclid.mdx add java and simplify

### PR DESCRIPTION
**Pull request by 876pol**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [ ] I have tested my code.
- [ ] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [ ] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making the changes.

### CODE REVIEW

File: content/6_Advanced/Extend_Euclid.mdx [hunk 1]
```diff
@@ -90,45 +110,70 @@ return more information!
...
-	for (;y[2];swap(x,y)) { // run extended Euclidean algo
-		ll k = x[2]/y[2];
-		F0R(i,3) x[i] -= k*y[i];
-		// keep subtracting multiple of one equation from the other
-	}
-	return x; // x[0]*a+x[1]*b=x[2], x[2]=gcd(a,b)
+array<ll, 3> extendEuclid(ll a, ll b) {
+    // we know that 1 * a + 0 * b = a and 0 * a + 1 * b = b
+    array<ll, 3> x = {1, 0, a};
+    array<ll, 3> y = {0, 1, b};
```
Comment (SansPapyrus683) [inline content/6_Advanced/Extend_Euclid.mdx:116]: tabs

**Comment by SansPapyrus683** [comment]
bump

File: content/6_Advanced/Extend_Euclid.mdx (hunk 1 above)
Comment (876pol) [reply content/6_Advanced/Extend_Euclid.mdx]: fixed

File: content/6_Advanced/Extend_Euclid.mdx [hunk 2]
```diff
@@ -145,12 +190,24 @@ def extend_euclid(a: int, b: int) -> list[int]:
...
 }
 ```
 
 </CPPSection>
+
+<JavaSection>
+
+```java
+static long euclid(long a, long b) {
+	if (b == 0) return a;
```
Comment (SansPapyrus683) [inline content/6_Advanced/Extend_Euclid.mdx:204]: braces

File: content/6_Advanced/Extend_Euclid.mdx [hunk 3]
```diff
@@ -171,12 +228,27 @@ becomes
...
 ```
 
 </CPPSection>
+
+<JavaSection>
+
+```java
+static long[] extendEuclid(long a, long b) { // returns {x,y}
+	if (b == 0)
+		return new long[] { 1, 0 };
```
Comment (SansPapyrus683) [inline content/6_Advanced/Extend_Euclid.mdx:244]: braces

File: content/6_Advanced/Extend_Euclid.mdx [hunk 4]
```diff
@@ -221,22 +293,50 @@ $n^2 < \texttt{LLONG\_MAX}$.
...
+<JavaSection>
+
+```java
+static long invGeneral(long a, long b) {
+	long[] x = extendEuclid(a, b);
+	assert (x[2] == 1); // gcd must be 1
+	return x[0] + (x[0] < 0 ? 1 : 0) * b;
+}
+
+public static void main(String[] args) {
```
Comment (SansPapyrus683) [inline content/6_Advanced/Extend_Euclid.mdx:325]: given that there's a main, i'd just put this in a class

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: content/6_Advanced/Extend_Euclid.mdx (hunk 4 above)
Comment (876pol) [reply content/6_Advanced/Extend_Euclid.mdx]: i'm just gonna remove the main function -> i don't think a test function is necessary

File: content/6_Advanced/Extend_Euclid.mdx [hunk 5]
```diff
@@ -220,41 +299,35 @@ $n^2 < \texttt{LLONG\_MAX}$.
...
-	}
-}
+</CPPSection>
+
+<JavaSection>
 
+```java
+static long invGeneral(long a, long b) {
+	long[] x = extendEuclid(a, b);
+	assert (x[2] == 1); // gcd must be 1
```
Comment (SansPapyrus683) [inline content/6_Advanced/Extend_Euclid.mdx:316]: two spaces after code for inline comment\
```suggestion
	assert (x[2] == 1);  // gcd must be 1
```

**Review by SansPapyrus683 — APPROVED** [review]
seems good

**Comment by SansPapyrus683** [comment]
@ryanchou-dev review & i'll merge

File: content/6_Advanced/Extend_Euclid.mdx [hunk 6]
```diff
@@ -90,45 +110,70 @@ return more information!
...
 
 ```cpp
-array<ll,3> extendEuclid(ll a, ll b) {
-	array<ll,3> x = {1,0,a}, y = {0,1,b};
-	// we know that 1*a+0*b=a and 0*a+1*b=b
-	for (;y[2];swap(x,y)) { // run extended Euclidean algo
-		ll k = x[2]/y[2];
-		F0R(i,3) x[i] -= k*y[i];
+array<ll, 3> extend_euclid(ll a, ll b) {
+	// we know that 1 * a + 0 * b = a and 0 * a + 1 * b = b
```
Comment (ryanchou-dev) [inline content/6_Advanced/Extend_Euclid.mdx:114]: add parens? makes it easier to read

**Review by ryanchou-dev — APPROVED** [review]

File: content/6_Advanced/Extend_Euclid.mdx [hunk 7]
```diff
@@ -26,15 +26,35 @@ The original Euclidean Algorithm computes $\gcd(a,b)$ and looks like this:
 
 ```cpp
 ll euclid(ll a, ll b) {
-	for (; b; swap(a, b)) {
+	while (b > 0) {
 		ll k = a / b;
 		a -= k * b;
+		swap(a, b)
```
Comment (bqi343) [inline content/6_Advanced/Extend_Euclid.mdx:32]: this doesn't compile (also, what was wrong with the original version?)

Comment (876pol) [reply content/6_Advanced/Extend_Euclid.mdx]: wanted to keep it consistent with java and python

**Pull request merged**